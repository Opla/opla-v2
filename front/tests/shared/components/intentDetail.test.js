/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import IntentDetail from "shared/components/intentDetail";

describe("components/intentDetail", () => {
  const defaultProps = {
    intent: {},
    onSelect: () => {},
    onEdit: () => {},
    onAction: () => {},
    onDisable: () => {},
    newActions: {
      input: undefined,
      output: undefined,
    },
  };

  const populatedIntent = {
    id: "T4Zs",
    botId: "HZAw",
    name: "Barquette alimentaire",
    input: ["*barquette", "barquette*", "*barquette*"],
    output: ["Attention, les barquettes alimentaires ne peuvent..."],
    order: 7,
  };

  it("renders correctly", () => {
    const intent = { topic: "some intent" };
    const component = renderer.create(
      <IntentDetail {...defaultProps} intent={intent} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render input intents", () => {
    const wrapper = shallow(
      <IntentDetail {...defaultProps} intent={populatedIntent} />,
    );

    expect(wrapper.find("ActionsList")).toHaveLength(2);
    /* Mik debug test : TODO 
    expect(
      wrapper
        .find("ActionsList")
        .at(0)
        .dive()
        .find("ActionsItem"),
    ).toHaveLength(4);
    expect(
      wrapper
        .find("ActionsList")
        .at(0)
        .dive()
        .find("ActionsItem")
        .at(2)
        .prop("action"),
    ).toEqual(populatedIntent.input[2]); */
  });

  it("Should render output intents", () => {
    const wrapper = shallow(
      <IntentDetail {...defaultProps} intent={populatedIntent} />,
    );

    expect(wrapper.find("ActionsList")).toHaveLength(2);
    expect(
      wrapper
        .find("ActionsList")
        .at(1)
        .dive()
        .find("ActionsItem"),
    ).toHaveLength(2);
    /* Mik debug test : TODO 
     expect(
      wrapper
        .find("ActionsList")
        .at(1)
        .dive()
        .find("ActionsItem")
        .at(0)
        .prop("action"),
    ).toEqual(populatedIntent.output[0]); */
  });

  it("can render intent with special characters correctly", () => {
    const intentcontent = [
      `{{${encodeURIComponent("{{text}}")}}}`,
      `<<${encodeURIComponent("<<var>>")}>>`,
    ].join("");
    const intent = { topic: intentcontent };
    const component = renderer.create(
      <IntentDetail {...defaultProps} intent={intent} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should call onSelect on Topic parameters click", () => {
    const onSelectSpy = jest.fn();
    const callArg = {
      name: populatedIntent.name,
      state: "topic",
    };
    const wrapper = shallow(
      <IntentDetail
        {...defaultProps}
        intent={populatedIntent}
        onSelect={onSelectSpy}
      />,
    );

    expect(wrapper.find("ExpansionPanel")).toHaveLength(1);

    const buttonsWrapper = wrapper
      .find("ExpansionPanel")
      .dive()
      .find("Button");
    expect(buttonsWrapper).toHaveLength(2);

    buttonsWrapper.at(0).simulate("click", { preventDefault: () => {} });

    expect(onSelectSpy).toHaveBeenCalled();
    expect(onSelectSpy.mock.calls[0][0]).toEqual(callArg);
  });
});
