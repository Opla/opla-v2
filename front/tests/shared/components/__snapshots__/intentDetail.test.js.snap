// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/intentDetail can render intent with special characters correctly 1`] = `
<div
  className="zui-action-panel list-box"
>
  <div
    className="list-content"
  >
    
    <div
      className="zui-expansion zui-color--white"
      style={
        Object {
          "margin": "12px",
        }
      }
    >
      <div
        className="zui-expansion-header"
        onClick={[Function]}
        style={
          Object {
            "height": "64px",
            "paddingTop": "4px",
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        <div
          className="zui-expansion-arrow"
        >
          <i
            className="zui-expansion-arrow-icon material-icons"
            disabled={false}
            style={
              Object {
                "animation": null,
                "fontSize": "24px",
                "paddingTop": "8px",
                "textAlign": "center",
              }
            }
          >
            keyboard_arrow_up
          </i>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "width": "100%",
            }
          }
        >
          <div
            className="zui-expansion-title"
            style={null}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "fontWeight": "900",
                }
              }
            >
              <i
                className="material-icons"
                style={
                  Object {
                    "marginLeft": "-8px",
                    "paddingRight": "4px",
                    "paddingTop": "12px",
                  }
                }
              >
                account_circle
              </i>
              input
              <i
                className="help_icon material-icons"
                onClick={[Function]}
                onKeyUp={[Function]}
                role="button"
                tabIndex="0"
              >
                help
              </i>
            </div>
          </div>
          
        </div>
      </div>
      <div
        className="zui-expansion-content"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "overflow": "hidden",
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        
        <li
          className="addActionItem  onFocusAction actionItem mdc-list-item"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <div
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              aria-label="Add an input sentence"
              className="contenteditable"
              id="ae_content"
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              spellCheck={false}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flex": 1,
                  "minHeight": "40px",
                }
              }
              tabIndex={0}
            >
              <span
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexWrap": "wrap",
                  }
                }
              >
                <span
                  className={undefined}
                  data="text"
                  id="ae_start"
                  onClick={[Function]}
                  onInput={[Function]}
                  style={
                    Object {
                      "display": "inline-block",
                      "flex": "1",
                      "minHeight": "15px",
                      "padding": "0 4px",
                    }
                  }
                  tabIndex={1}
                />
              </span>
              <div>
                Add an input sentence
              </div>
            </div>
          </div>
          <div />
        </li>
        <ul
          className="mdc-list"
          style={
            Object {
              "margin": "0 8px 16px 8px",
              "maxHeight": "18.07vh",
              "overflowX": "hidden",
              "overflowY": "scroll",
            }
          }
        />
      </div>
    </div>
    <div
      className="zui-expansion zui-color--white"
      style={
        Object {
          "margin": "12px",
        }
      }
    >
      <div
        className="zui-expansion-header"
        onClick={[Function]}
        style={
          Object {
            "height": "64px",
            "paddingTop": "4px",
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        <div
          className="zui-expansion-arrow"
        >
          <i
            className="zui-expansion-arrow-icon material-icons"
            disabled={false}
            style={
              Object {
                "animation": null,
                "fontSize": "24px",
                "paddingTop": "8px",
                "textAlign": "center",
              }
            }
          >
            keyboard_arrow_up
          </i>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "width": "100%",
            }
          }
        >
          <div
            className="zui-expansion-title"
            style={null}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "fontWeight": "900",
                }
              }
            >
              <i
                className="material-icons"
                style={
                  Object {
                    "backgroundImage": "url(../images/opla-avatar.png)",
                    "backgroundPositionX": "2px",
                    "backgroundPositionY": "14px",
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "20px",
                    "color": "transparent",
                    "marginLeft": "-8px",
                    "paddingRight": "4px",
                    "paddingTop": "12px",
                  }
                }
              >
                account_circle
              </i>
              output
              <i
                className="help_icon material-icons"
                onClick={[Function]}
                onKeyUp={[Function]}
                role="button"
                tabIndex="0"
              >
                help
              </i>
            </div>
          </div>
          
        </div>
      </div>
      <div
        className="zui-expansion-content"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "overflow": "hidden",
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        
        <li
          className="addActionItem  onFocusAction actionItem mdc-list-item"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <div
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              aria-label="Add an output response"
              className="contenteditable"
              id="ae_content"
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              spellCheck={false}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flex": 1,
                  "minHeight": "40px",
                }
              }
              tabIndex={0}
            >
              <span
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexWrap": "wrap",
                  }
                }
              >
                <span
                  className={undefined}
                  data="text"
                  id="ae_start"
                  onClick={[Function]}
                  onInput={[Function]}
                  style={
                    Object {
                      "display": "inline-block",
                      "flex": "1",
                      "minHeight": "15px",
                      "padding": "0 4px",
                    }
                  }
                  tabIndex={1}
                />
              </span>
              <div>
                Add an output response
              </div>
            </div>
          </div>
          <div />
        </li>
        <ul
          className="mdc-list"
          style={
            Object {
              "margin": "0 8px 16px 8px",
              "maxHeight": "18.07vh",
              "overflowX": "hidden",
              "overflowY": "scroll",
            }
          }
        />
      </div>
    </div>
    <div
      className="zui-expansion zui-color--white"
      style={
        Object {
          "margin": "12px",
        }
      }
    >
      <div
        className="zui-expansion-header"
        onClick={[Function]}
        style={
          Object {
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        <div
          className="zui-expansion-arrow"
        >
          <i
            className="zui-expansion-arrow-icon material-icons"
            disabled={false}
            style={
              Object {
                "animation": null,
                "fontSize": "24px",
                "paddingTop": "8px",
                "textAlign": "center",
              }
            }
          >
            keyboard_arrow_down
          </i>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "width": "100%",
            }
          }
        >
          <div
            className="zui-expansion-title"
            style={null}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "fontWeight": "900",
                }
              }
            >
              <i
                className="zrmc-icon"
                style={
                  Object {
                    "height": "24px",
                    "marginLeft": "-20px",
                    "marginRight": "16px",
                    "marginTop": "0px",
                    "width": "24px",
                  }
                }
              >
                <svg>
                  <path
                    d="M7.75,13C7.74,12.65 7.9,12.31 8.17,12.08C8.92,12.24 9.62,12.55 10.25,13C10.25,13.68 9.69,14.24 9,14.24C8.31,14.24 7.76,13.69 7.75,13M13.75,13C14.38,12.56 15.08,12.25 15.83,12.09C16.1,12.32 16.26,12.66 16.25,13C16.25,13.7 15.69,14.26 15,14.26C14.31,14.26 13.75,13.7 13.75,13V13M12,9C9.23,8.96 6.5,9.65 4.07,11L4,12C4,13.23 4.29,14.44 4.84,15.54C7.21,15.18 9.6,15 12,15C14.4,15 16.79,15.18 19.16,15.54C19.71,14.44 20,13.23 20,12L19.93,11C17.5,9.65 14.77,8.96 12,9M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2Z"
                    fill="#000000"
                  />
                </svg>
              </i>
              Advanced
              <i
                className="help_icon material-icons"
                onClick={[Function]}
                onKeyUp={[Function]}
                role="button"
                tabIndex="0"
              >
                help
              </i>
            </div>
          </div>
          
        </div>
      </div>
      <div
        className="zui-expansion-content"
        style={
          Object {
            "display": "block",
            "height": "0px",
            "overflow": "hidden",
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        <ul
          className="mdc-list"
        >
          <li
            className="mdc-list-item"
            onKeyPress={[Function]}
            role="menuitem"
          >
            <i
              aria-hidden="true"
              className="mdc-list-item__graphic material-icons"
            >
              title
            </i>
            Topic
            <span
              className="mdc-list-item__meta"
              onClick={null}
              onKeyUp={[Function]}
              role="button"
              tabIndex="0"
            >
              <button
                className="zui-action-button mdc-button"
                disabled={false}
                id={null}
                onClick={[Function]}
                type="button"
              >
                
                {{%7B%7Btext%7D%7D}}&lt;&lt;%3C%3Cvar%3E%3E&gt;&gt;
              </button>
            </span>
          </li>
          <li
            className="mdc-list-item"
            onKeyPress={[Function]}
            role="menuitem"
          >
            <i
              aria-hidden="true"
              className="mdc-list-item__graphic material-icons"
            >
              link
            </i>
            Previous
            <span
              className="mdc-list-item__meta"
              onClick={null}
              onKeyUp={[Function]}
              role="button"
              tabIndex="0"
            >
              <button
                className="zui-action-button mdc-button"
                disabled={false}
                id={null}
                onClick={[Function]}
                type="button"
              >
                
                NONE
              </button>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/intentDetail renders correctly 1`] = `
<div
  className="zui-action-panel list-box"
>
  <div
    className="list-content"
  >
    
    <div
      className="zui-expansion zui-color--white"
      style={
        Object {
          "margin": "12px",
        }
      }
    >
      <div
        className="zui-expansion-header"
        onClick={[Function]}
        style={
          Object {
            "height": "64px",
            "paddingTop": "4px",
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        <div
          className="zui-expansion-arrow"
        >
          <i
            className="zui-expansion-arrow-icon material-icons"
            disabled={false}
            style={
              Object {
                "animation": null,
                "fontSize": "24px",
                "paddingTop": "8px",
                "textAlign": "center",
              }
            }
          >
            keyboard_arrow_up
          </i>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "width": "100%",
            }
          }
        >
          <div
            className="zui-expansion-title"
            style={null}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "fontWeight": "900",
                }
              }
            >
              <i
                className="material-icons"
                style={
                  Object {
                    "marginLeft": "-8px",
                    "paddingRight": "4px",
                    "paddingTop": "12px",
                  }
                }
              >
                account_circle
              </i>
              input
              <i
                className="help_icon material-icons"
                onClick={[Function]}
                onKeyUp={[Function]}
                role="button"
                tabIndex="0"
              >
                help
              </i>
            </div>
          </div>
          
        </div>
      </div>
      <div
        className="zui-expansion-content"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "overflow": "hidden",
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        
        <li
          className="addActionItem  onFocusAction actionItem mdc-list-item"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <div
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              aria-label="Add an input sentence"
              className="contenteditable"
              id="ae_content"
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              spellCheck={false}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flex": 1,
                  "minHeight": "40px",
                }
              }
              tabIndex={0}
            >
              <span
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexWrap": "wrap",
                  }
                }
              >
                <span
                  className={undefined}
                  data="text"
                  id="ae_start"
                  onClick={[Function]}
                  onInput={[Function]}
                  style={
                    Object {
                      "display": "inline-block",
                      "flex": "1",
                      "minHeight": "15px",
                      "padding": "0 4px",
                    }
                  }
                  tabIndex={1}
                />
              </span>
              <div>
                Add an input sentence
              </div>
            </div>
          </div>
          <div />
        </li>
        <ul
          className="mdc-list"
          style={
            Object {
              "margin": "0 8px 16px 8px",
              "maxHeight": "18.07vh",
              "overflowX": "hidden",
              "overflowY": "scroll",
            }
          }
        />
      </div>
    </div>
    <div
      className="zui-expansion zui-color--white"
      style={
        Object {
          "margin": "12px",
        }
      }
    >
      <div
        className="zui-expansion-header"
        onClick={[Function]}
        style={
          Object {
            "height": "64px",
            "paddingTop": "4px",
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        <div
          className="zui-expansion-arrow"
        >
          <i
            className="zui-expansion-arrow-icon material-icons"
            disabled={false}
            style={
              Object {
                "animation": null,
                "fontSize": "24px",
                "paddingTop": "8px",
                "textAlign": "center",
              }
            }
          >
            keyboard_arrow_up
          </i>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "width": "100%",
            }
          }
        >
          <div
            className="zui-expansion-title"
            style={null}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "fontWeight": "900",
                }
              }
            >
              <i
                className="material-icons"
                style={
                  Object {
                    "backgroundImage": "url(../images/opla-avatar.png)",
                    "backgroundPositionX": "2px",
                    "backgroundPositionY": "14px",
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "20px",
                    "color": "transparent",
                    "marginLeft": "-8px",
                    "paddingRight": "4px",
                    "paddingTop": "12px",
                  }
                }
              >
                account_circle
              </i>
              output
              <i
                className="help_icon material-icons"
                onClick={[Function]}
                onKeyUp={[Function]}
                role="button"
                tabIndex="0"
              >
                help
              </i>
            </div>
          </div>
          
        </div>
      </div>
      <div
        className="zui-expansion-content"
        style={
          Object {
            "display": "block",
            "height": "100%",
            "overflow": "hidden",
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        
        <li
          className="addActionItem  onFocusAction actionItem mdc-list-item"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <div
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <div
              aria-label="Add an output response"
              className="contenteditable"
              id="ae_content"
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              spellCheck={false}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flex": 1,
                  "minHeight": "40px",
                }
              }
              tabIndex={0}
            >
              <span
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexWrap": "wrap",
                  }
                }
              >
                <span
                  className={undefined}
                  data="text"
                  id="ae_start"
                  onClick={[Function]}
                  onInput={[Function]}
                  style={
                    Object {
                      "display": "inline-block",
                      "flex": "1",
                      "minHeight": "15px",
                      "padding": "0 4px",
                    }
                  }
                  tabIndex={1}
                />
              </span>
              <div>
                Add an output response
              </div>
            </div>
          </div>
          <div />
        </li>
        <ul
          className="mdc-list"
          style={
            Object {
              "margin": "0 8px 16px 8px",
              "maxHeight": "18.07vh",
              "overflowX": "hidden",
              "overflowY": "scroll",
            }
          }
        />
      </div>
    </div>
    <div
      className="zui-expansion zui-color--white"
      style={
        Object {
          "margin": "12px",
        }
      }
    >
      <div
        className="zui-expansion-header"
        onClick={[Function]}
        style={
          Object {
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        <div
          className="zui-expansion-arrow"
        >
          <i
            className="zui-expansion-arrow-icon material-icons"
            disabled={false}
            style={
              Object {
                "animation": null,
                "fontSize": "24px",
                "paddingTop": "8px",
                "textAlign": "center",
              }
            }
          >
            keyboard_arrow_down
          </i>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "width": "100%",
            }
          }
        >
          <div
            className="zui-expansion-title"
            style={null}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "fontWeight": "900",
                }
              }
            >
              <i
                className="zrmc-icon"
                style={
                  Object {
                    "height": "24px",
                    "marginLeft": "-20px",
                    "marginRight": "16px",
                    "marginTop": "0px",
                    "width": "24px",
                  }
                }
              >
                <svg>
                  <path
                    d="M7.75,13C7.74,12.65 7.9,12.31 8.17,12.08C8.92,12.24 9.62,12.55 10.25,13C10.25,13.68 9.69,14.24 9,14.24C8.31,14.24 7.76,13.69 7.75,13M13.75,13C14.38,12.56 15.08,12.25 15.83,12.09C16.1,12.32 16.26,12.66 16.25,13C16.25,13.7 15.69,14.26 15,14.26C14.31,14.26 13.75,13.7 13.75,13V13M12,9C9.23,8.96 6.5,9.65 4.07,11L4,12C4,13.23 4.29,14.44 4.84,15.54C7.21,15.18 9.6,15 12,15C14.4,15 16.79,15.18 19.16,15.54C19.71,14.44 20,13.23 20,12L19.93,11C17.5,9.65 14.77,8.96 12,9M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2Z"
                    fill="#000000"
                  />
                </svg>
              </i>
              Advanced
              <i
                className="help_icon material-icons"
                onClick={[Function]}
                onKeyUp={[Function]}
                role="button"
                tabIndex="0"
              >
                help
              </i>
            </div>
          </div>
          
        </div>
      </div>
      <div
        className="zui-expansion-content"
        style={
          Object {
            "display": "block",
            "height": "0px",
            "overflow": "hidden",
            "transition": "height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
          }
        }
      >
        <ul
          className="mdc-list"
        >
          <li
            className="mdc-list-item"
            onKeyPress={[Function]}
            role="menuitem"
          >
            <i
              aria-hidden="true"
              className="mdc-list-item__graphic material-icons"
            >
              title
            </i>
            Topic
            <span
              className="mdc-list-item__meta"
              onClick={null}
              onKeyUp={[Function]}
              role="button"
              tabIndex="0"
            >
              <button
                className="zui-action-button mdc-button"
                disabled={false}
                id={null}
                onClick={[Function]}
                type="button"
              >
                
                some intent
              </button>
            </span>
          </li>
          <li
            className="mdc-list-item"
            onKeyPress={[Function]}
            role="menuitem"
          >
            <i
              aria-hidden="true"
              className="mdc-list-item__graphic material-icons"
            >
              link
            </i>
            Previous
            <span
              className="mdc-list-item__meta"
              onClick={null}
              onKeyUp={[Function]}
              role="button"
              tabIndex="0"
            >
              <button
                className="zui-action-button mdc-button"
                disabled={false}
                id={null}
                onClick={[Function]}
                type="button"
              >
                
                NONE
              </button>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
